src/iotjs_binding.cpp

```diff
@@ -472,178 +470,320 @@ bool JResult::IsException() const {
 }
 
 
-JRawValueType JVal::Undefined() {
+} // namespace iotjs
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+JRawValueType iotjs_jval_undefined() {
   return jerry_create_undefined();
 }
 
 
-JRawValueType JVal::Null() {
+JRawValueType iotjs_jval_null() {
   return jerry_create_null();
 }
 
 
-JRawValueType JVal::Bool(bool v) {
+JRawValueType iotjs_jval_bool(bool v) {
   return jerry_create_boolean(v);
 }
 
 
-JRawValueType JVal::Number(int v) {
-  return JVal::Number((double)v);
+JRawValueType iotjs_jval_number(double v) {
+  return jerry_create_number(v);
 }
 
 
-JRawValueType JVal::Number(double v) {
-  return jerry_create_number(v);
+JRawValueType iotjs_jval_string(const iotjs_string_t* v) {
+  return jerry_create_string_sz(
+      reinterpret_cast<const jerry_char_t*>(iotjs_string_data(v)),
+                                            iotjs_string_size(v));
+
 }
 
 
-JRawValueType JVal::Object() {
+JRawValueType iotjs_jval_raw_string(const char* v) {
+  IOTJS_ASSERT(v != NULL);
+  return jerry_create_string(reinterpret_cast<const jerry_char_t*>(v));
+}
+
+
+
+JRawValueType iotjs_jval_object() {
   return jerry_create_object();
 }
 
 
-JArgList::JArgList(uint16_t capacity)
-    : _capacity(capacity)
-    , _argc(0)
-    , _argv(NULL) {
-  if (_capacity > 0) {
-    _argv = new JObject*[_capacity];
-    for (int i = 0; i < _capacity; ++i) {
-      _argv[i] = NULL;
-    }
+iotjs_jargs_t iotjs_jargs_create(uint16_t capacity) {
+  iotjs_jargs_t jargs;
+  IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_jargs_t, &jargs);
+
+  _this->capacity = capacity;
+  _this->argc = 0;
+  if (capacity > 0) {
+    unsigned buffer_size = sizeof(void*) * capacity;
+    _this->argv = (iotjs::JObject**) iotjs_buffer_allocate(buffer_size);
+  } else {
+    _this->argv = NULL;
   }
+
+  return jargs;
 }
 
 
-JArgList::~JArgList() {
-  IOTJS_ASSERT(_argv == NULL || _argc > 0);
-  IOTJS_ASSERT(_argc <= _capacity);
-  for (int i = 0; i < _argc; ++i) {
-    delete _argv[i];
+void iotjs_jargs_destroy(iotjs_jargs_t* jargs) {
+  IOTJS_VALIDATED_STRUCT_DESTRUCTOR(iotjs_jargs_t, jargs);
+
+  IOTJS_ASSERT(_this->argv == NULL || _this->argc > 0);
+  IOTJS_ASSERT(_this->argc <= _this->capacity);
+
+  if (_this->capacity > 0) {
+    for (int i = 0; i < _this->argc; ++i) {
+      delete _this->argv[i];
```

Comment:
Yes it is intentional. The type of `_this->argv[i]` is still `iotjs::JObject*` (c++ class), so I didn't removed the `delete` operator yet. I'll fix it after I change `class iotjs::JObject` into `struct iotjs_jobject_t`.